benches/state_properties.rs

```diff
@@ -0,0 +1,144 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use feos::pcsaft::{PcSaft, PcSaftParameters};
+use feos_core::{
+    parameter::{IdentifierOption, Parameter},
+    Contributions, State,
+};
+use ndarray::{arr1, Array1};
+use quantity::si::*;
+use std::sync::Arc;
+
+fn helmholtz_energy(inp: (&Arc<PcSaft>, SINumber, SINumber, &SIArray1)) -> f64 {
+    State::new(
+        inp.0,
+        Some(inp.1),
+        Some(inp.2),
+        None,
+        None,
+        None,
+        Some(inp.3),
+        None,
+        None,
+        None,
+        None,
+        None,
+        feos_core::DensityInitialization::None,
+        None,
+    )
+    .unwrap()
+    .compressibility(Contributions::Total)
+}
+
+fn compressibility(inp: (&Arc<PcSaft>, SINumber, SINumber, &Array1<f64>)) -> f64 {
+    State::new(
+        inp.0,
+        Some(inp.1),
+        None,
+        Some(inp.2),
+        None,
+        None,
+        None,
+        Some(inp.3),
+        None,
+        None,
+        None,
+        None,
+        feos_core::DensityInitialization::None,
+        None,
+    )
+    .unwrap()
+    .compressibility(Contributions::Total)
+}
+
+fn ln_phi(inp: (&Arc<PcSaft>, SINumber, SINumber, &Array1<f64>)) -> Array1<f64> {
```

Comment:
```suggestion
fn property<T, F: Fn(&State<PcSaft, SIUnit>) -> T(f: F, inp: (&Arc<PcSaft>, SINumber, SINumber, &Array1<f64>)) -> T {
    f(State::new(...))
}
```
maybe something like that?